modules/commons/src/main/java/org/apache/synapse/commons/staxon/core/base/AbstractXMLInputFactory.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2011, 2012 Odysseus Software GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.synapse.commons.staxon.core.base;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import javax.xml.stream.StreamFilter;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLReporter;
+import javax.xml.stream.XMLResolver;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import javax.xml.stream.util.StreamReaderDelegate;
+import javax.xml.stream.util.XMLEventAllocator;
+import javax.xml.transform.Source;
+import javax.xml.transform.stream.StreamSource;
+
+/**
+ * Abstract XML input factory.
+ */
+public abstract class AbstractXMLInputFactory extends XMLInputFactory {
+    private XMLEventAllocator allocator;
+    private XMLResolver resolver;
+    private XMLReporter reporter;
+
+    private boolean coalescing;
+    private boolean namespaceAware;
+    private boolean replacingEntityReferences;
+    private boolean supportingExternalEntities;
+    private boolean validating;
+    private boolean supportDTD;
+
+    @Override
+    public XMLStreamReader createXMLStreamReader(Source source) throws XMLStreamException {
+        if (source instanceof StreamSource) {
+            StreamSource streamSource = (StreamSource) source;
+            InputStream input = streamSource.getInputStream();
+            if (input != null) {
+                if (streamSource.getSystemId() != null) {
+                    return createXMLStreamReader(streamSource.getSystemId(), input);
+                } else {
+                    return createXMLStreamReader(input);
+                }
+            }
+            Reader reader = streamSource.getReader();
+            if (reader != null) {
+                if (streamSource.getSystemId() != null) {
+                    return createXMLStreamReader(streamSource.getSystemId(), reader);
+                } else {
+                    return createXMLStreamReader(reader);
+                }
+            }
+            if (streamSource.getSystemId() != null) {
+                try {
+                    final InputStream stream = new URI(source.getSystemId()).toURL().openStream();

```

Comment:
Does the caller of this method need to close the stream after using it? If so might be better to mention it in the method comment.